moml/data/pfas_sdf_dataset.py

```
@@ -26,168 +178,440 @@ def __init__(
-                len([a for a in mol.GetAtoms() if a.GetSymbol() == 'F'])  # Fluorine count (PFAS-specific)
+                Descriptors.LabuteASA(mol),                # Labute accessible surface area
+                Descriptors.NumHeteroatoms(mol),           # Heteroatom count
+                self._count_fluorine_atoms(mol),           # Fluorine count (PFAS-specific)
             ]
             
-            # Handle any None or invalid values
-            descriptors = [float(d) if d is not None and not np.isnan(float(d)) else 0.0 
-                          for d in descriptors]
+            # Validate and clean descriptor values
+            clean_descriptors = []
+            for i, desc in enumerate(descriptors):
```

Ruff (0.12.2)

Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)